content/post/event-triggered-customer-segmentation.md

```diff
@@ -0,0 +1,247 @@
+    	   		    info[2].strip(), info[3].strip())
+
+def build_purchase_conversion(info):
+    return PurchaseConversion(info[0].strip(), info[1].strip(),
+    	   		      info[2].strip(), info[3].strip(),
+			      info[3].strip())
+```
+The next step in the pipeline will process your conversions data.  
+```
+ab.to_state_partition(process_email_add_customer, LoyaltyCustomers,
+    "loyalty customers", extract_conversion_key, initial_partitions)
+```
+Similarly to your first pipeline, you'll call the `process_email_add_customer` function using the `LoyaltyCustomers` state object, using the same conversion keys and initial partitions as last time. The `process_email_add_customer` function actually handles two things: determining whether to send the extra 10% off to your loyalty customers when they've clicked your ad, and adding a new customer as a loyalty customer once they've completed a purchase. You'll need to send on both the full conversion information, as well as the `should_email` information to your next processor, for the email to be sent. So you'll need to return a tuple within a tuple. 
+```
+@wallaroo.state_computation(name="add_new_loyalty_customer")
+def process_email_add_customer(conversion, loyalty_customers):
+    should_email = False
+    if conversion.customer_email in loyalty_customers.customers:
+        if conversion.conversion == 'ClickConversion':
+            should_email = True
+    else:
+        if conversion.conversion == 'PurchaseConversion':
+            loyalty_customers.add(conversion.customer_email,
+	        Customer(conversion.customer_email, '','',''))
+    return ((conversion, should_email), True)
+```
+This time, rather than calling `ab.done()`, you'll want to send the data on to a sink. Wallaroo needs to know how to reach it. You'll also need to create a new encoder function.
+```
+ab.to_sink(wallaroo.TCPSinkConfig(out_host, out_port, cc_encoder))
+```
+Your encoder message can be quite simple. Here, you'll quickly unpack the tuple that was sent from the previous pipeline step, and format that into a string. You might also want to send a message out to, for example, Kafka, for further processing. This function ensures that the message is formatted correctly for your next step. 
+```
+@wallaroo.encoder
+def cc_encoder(data):
+    (conversion, should_email) = data
+    return ("Send additional promo code to: " + conversion.customer_email
+            + ", " + str(should_email) + "\n")
+```
+This is the last step in the conversions pipeline. So, let's return the `ApplicationBuilder`, and you're done! 
+```
+return ab.build()
+```
+## Conclusions
+In just over 120 lines of code, you've quickly and smoothly created a lightweight stream-processed application to manage an aspect of your ad campaigns. We'll be looking at more examples in future blog posts of small, quick applications that you can add on to your current setup. At Wallaroo, we're just excited to help you be more effective! 
```

Comment:
Usually we close with a call to action kind of message that tells folks where to find more information and invites them to check out our github and other resources. I think there should be something here that suggests what the reader could do next.